### PR TITLE
Bug 1695884 - Put PingType in a standalone file (#)

### PR DESCRIPTION
**Pull request by ChinYing-Li**
This PR aims to fix [bug 1695884](https://bugzilla.mozilla.org/show_bug.cgi?id=1695884).
Any suggestion is appreciated!

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 * [#173](https://github.com/mozilla/glean.js/pull/173): Drop Node.js support from webext entry points
 * [#155](https://github.com/mozilla/glean.js/pull/155): Allow to define custom uploaders in the configuration.
+* [#183](https://github.com/mozilla/glean.js/pull/183): Put PingType in a standalone file
```
Comment (brizental) [inline CHANGELOG.md:7]: Not necessary to put this in the CHANGELOG.md file as it is just an internal change.

File: glean/src/core/internal_pings.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import { DELETION_REQUEST_PING_NAME } from "./constants.js";
-import PingType from "./pings/index.js";
+import PingType from "./pings/pingtype.js";
```
Comment (brizental) [inline glean/src/core/internal_pings.ts:6]: Let's please use snake_case for file names to keep consistency i.e. `ping_type`.

File: glean/src/core/pings/index.ts [hunk 3]
```diff
@@ -21,64 +16,4 @@ interface CommonPingData {
...
-  
-      const identifier = generateUUIDv4();
-      await collectAndStorePing(identifier, this, correctedReason);
-      return;
-    });
-  }
-}
-
-export default PingType;
+export default CommonPingData;
```
Comment (brizental) [inline glean/src/core/pings/index.ts:19]: I know CommonPingData was moved to another file in a previous PR. Would you mind rebasing your PR to include that change? You'll probably get some merge conflicts when you do that.

**Review by brizental — CHANGES_REQUESTED** [review]
Hey @ChinYing-Li thank you for your contribution. This is looking great. I left a few comments to be addressed in the code.

File: glean/src/core/pings/index.ts [hunk 4]
```diff
@@ -1,80 +1,3 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (brizental) [inline glean/src/core/pings/index.ts:3]: since this file doesn't have anything on it anymore, feel free to delete it.

**Review by brizental — CHANGES_REQUESTED** [review]

**Comment by ChinYing-Li** [comment]
@brizental After rebasing, the PR is ready for review. Any suggestion is appreciated!
There's a conflict with `main` since I removed `glean/src/core/pings/index.ts`.

**Comment by brizental** [comment]
Changes look good, but you need to resolve the conflict before I can merge :)

**Comment by ChinYing-Li** [comment]
Closing this PR to open a clean one, as I don't have a quick solution to the conflicting index.ts; sorry for the inconvenience caused!

**Comment by brizental** [comment]
Looks great! Thanks for the hard work in cleaning up your commit history, great job :) Note that you didn't need to close the PR, force pushing would have been enough.

**Review by brizental — APPROVED** [review]

**Pull request merged**